.github/workflows/cicd.yml

```
@@ -0,0 +1,76 @@
+name: Check, Test, Build, Release
+on:
+  push:
+    branches:
+      - main
+      
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: [ 1.58.1 ]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache Cargo
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ matrix.build }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ matrix.build }}-cargo-
+            
+      - name: Install Rust
+        run: |
+          rustup update ${{ matrix.rust }} --no-self-update
+          rustup default ${{ matrix.rust }}
+          rustup component add rustfmt
+          rustup component add clippy
+          
+      - name: Check Format
+        run: cargo fmt --all -- --check
+
+      - name: Clippy
+        run: cargo clippy --tests --benches -- -D clippy::all
+
+      - name: Test
+        run: cargo test
+        env:
+          RUST_BACKTRACE: 1
+   
+      - name: Compute diff between local and upstream candid files
+        run: |
+          curl -s https://raw.githubusercontent.com/dfinity/cdk-rs/main/src/ic-certified-assets/assets.did > upstream_assets.did
+          echo CANDID_DIFF=$(diff upstream_assets.did assets.did | wc -l) >> $GITHUB_ENV
```

What would you think about displaying the diff here, and capturing the return code (which will be nonzero if there were differences) for later comparison?  Something like
```suggestion
          diff upstream_assets.did assets.did
          echo CANDID_DIFF_RC="$?" >> $GITHUB_ENV
```

This will help troubleshooting when there is a difference.

---

btw, what does `_RC` stand for here?

---

Return Code.  But name it what you want: could be `CANDID_DIFF_RESULT` or something else!